Lib/ipaddress.py

```diff
@@ -892,10 +884,9 @@ def compare_networks(self, other):
               TypeError if the IP versions are different.
 
         """
-        # does this need to raise a ValueError?
-        if self.version != other.version:
-            raise TypeError('%s and %s are not of the same type' % (
-                             self, other))
+        if not isinstance(other, _BaseNetwork):
+            raise TypeError("%s is not a network object" % other)
```

Comment:
This will raise weird error if `other` is a tuple. There are many such cases here.

Also, I doubt it is right to include the argument _value_ instead of _type_ in the error message. Well, it may work if it is a part of the `_BaseAddress` hierarhy, but if it is an arbitrary object, its repr can be arbitrary large.